Share contact code

vitvly opened the pull request:
Fix #4438 , #4992 

This fix implements "Share contact code" window. Temporarily not in a modal, but in a right sidebar pane (due to https://github.com/status-im/react-native-desktop/issues/258)

Commit:
Add share-contact-code

Qr code copy window

Copy code on-press handler

Add profile styles

Alignments, obsolete style removed

Add qr code close icon

Fix typo

Add copy-to-clipboard event handler and fx

Remove obsolete current-account binding

Formatting

Remove unneeded horizontal lines

Center qr code text

Center 'Share my contact code' button

Add green 'Copied to clipboard' tooltip view

flexsurfer commented on src/status_im/ui/screens/desktop/main/tabs/profile/styles.cljs:
```diff
@@ -1,5 +1,78 @@
+
+(def qr-code-copy-text
+  {:font-size 16
+   :color colors/white})
```
indentation

flexsurfer commented on src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:
```diff
@@ -25,6 +29,34 @@
      value]]]])
 
+(views/defview qr-code []
+  (views/letsubs [current-account [:get-current-account]]
```
`{:keys [public-key]} [:get-current-account]`

flexsurfer commented on src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:
```diff
@@ -25,6 +29,34 @@
+        [react/text {:style styles/qr-code-title}
+         (string/replace (i18n/label :qr-code-public-key-hint) "\n" "")]
+        [react/view {:style styles/qr-code}
+         [qr-code-viewer/qr-code {:value public-key :size 130} #_{:style styles/qr-code}]]
```
could you remove commented part or add TODO

flexsurfer commented on src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:
```diff
@@ -25,6 +29,34 @@
+         [qr-code-viewer/qr-code {:value public-key :size 130} #_{:style styles/qr-code}]]
+        [react/text {:style styles/qr-code-text}
+         public-key]
+        [react/touchable-highlight {:on-press #(react/copy-to-clipboard public-key)}
```
better to move `react/copy-to-clipboard` to effect, dispatch event and use custom fx in event

flexsurfer replied:
views should be pure

flexsurfer commented on src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:
```diff
@@ -38,7 +70,8 @@
      [react/view
-      [my-profile-info current-account]]
+      [share-contact-code]
+      #_[my-profile-info current-account]]
```
please remove or add TODO

vitvly commented:
Thank you for review @flexsurfer ! I've made the requested changes, and moved the clipboard copy functionality to the custom fx handler.

flexsurfer commented on src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:
```diff
@@ -18,34 +18,22 @@
-         [react/view {:style styles/qr-code-copy}
-          [react/text {:style styles/qr-code-copy-text}
-           (i18n/label :copy-qr)]]]]])))
+  (views/letsubs [{:keys [public-key] :as current-account} [:get-current-account]]
```
but you don't use `current-account` why do you need it? just wondering

vitvly replied:
sorry, you're right, fixed it, thought it was referred to in the body of fn.

rcullito commented on src/status_im/ui/screens/profile/events.cljs:
```diff
@@ -113,3 +113,13 @@
+
+(re-frame/reg-fx
+  :copy-to-clipboard
+  (fn [value]
```
is there any interceptor, etc, that causes the value to be the first parameter here instead of the event name?

flexsurfer replied:
it's effect, not event https://github.com/Day8/re-frame/blob/master/docs/Effects.md#extensible-side-effects

rcullito replied:
ah I see. completely new effect being introduced. thanks for the link.

rcullito reviewed (approved):
nice workaround for the modal issue vitaliy! 

does desktop have any current working implementation that you've seen for this type of component, or are we all the way down to the popup?

EugeOrtiz commented:
I cannot find how to see this in the latest builds, should I download a special one? Or click on something? 🤔

maxhora commented:
@EugeOrtiz you should check Jenkins build status for it (Click on `Details` for Jenkins check in the bottom of PR page), but since it's in failed state the build for this PR is not available yet.

churik commented:
Screen (all issues are marked):
![shareqr](https://user-images.githubusercontent.com/4557972/42079972-daf6f242-7b89-11e8-8699-777bae8d8689.png)

- [x] (1) No QR is displayed
- [x] (2) After copying contact code no message "Copied to clipboard" (according to zpl.io/VxY83em )
- [x] (3) Text is not centered
- [x] (4, 5) Grey lines are displayed on top and bottom button edges; "Share my contact code" button is not centered.
- [x] (6) "Close" button is not working (is it not  status-im/react-native-desktop#258 ?)

vitvly commented:
1 - Weird, cannot reproduce.
3, 4, 5 - fixed
6 - yes, close button there is not functional because there is nothing to close (there is no modal)

EugeOrtiz commented:
@churik some of your indications are also here #4986

EugeOrtiz commented:
Adding:
- [x] "Share my contact code" button should be aligned in the center of the left panel:
<img width="283" alt="captura de pantalla 2018-06-28 a la s 4 12 31 p m" src="https://user-images.githubusercontent.com/37297779/42055639-196dc236-7aee-11e8-81bf-ff18c3eee202.png">

- [x] Close icon ("X") shouldn't be displayed if it doesn't work.
- [x] Right panel should disappear (if it cannot be closed) when I navigate to "Home" tab.

Commit:
Add fading for qr code copied tooltip

Commit:
Always clear intervals

Commit:
Change profile view styling

Commit:
Refresh right pane when navigating to a different tab

Commit:
Increase left pane width to 340

vitvly commented:
#2 - added a tooltip that fades away after a couple of seconds
Removed close button.
Also changed profile view in order to address #4992 .
Right panel disappears when user navigates to a different tab.

churik commented:
Created #5098, #5099
